airtime_mvc/application/controllers/plugins/Acl_plugin.php

```diff
@@ -148,17 +148,22 @@ public function preDispatch(Zend_Controller_Request_Abstract $request)
                 }
             }
         } else { //We have a session/identity.
-
             // If we have an identity and we're making a RESTful request,
             // we need to check the CSRF token
-            if ($request->_action != "get" && $request->getModuleName() == "rest") {
-                $tokenValid = $this->verifyCSRFToken($request->getParam("csrf_token"));
+            if ($_SERVER['REQUEST_METHOD'] != "GET" && $request->getModuleName() == "rest") {
+                $token = $request->getParam("csrf_token");
+                $tokenValid = $this->verifyCSRFToken($token);
 
                 if (!$tokenValid) {
+                    $csrf_namespace = new Zend_Session_Namespace('csrf_namespace');
+                    $csrf_namespace->authtoken = sha1(uniqid(rand(),1));
```

Comment:
Both the rand() and uniqid() docs say that these are not crytographically secure:
http://php.net/manual/en/function.rand.php
http://php.net/manual/en/function.uniqid.php

Can you try using this function instead?
http://php.net/manual/en/function.openssl-random-pseudo-bytes.php